ncs/encrypt_script.py

```diff
@@ -100,44 +67,31 @@ def __init__(self, hash_name: str):
             raise ValueError(f"Unsupported hash algorithm: {hash_name}")
         self._hash_name = hash_name
 
-    def generate_digest_size_for_plain_text(self, plaintext_file_path: Path, output_directory: Path):
+    def generate_digest_size_for_plain_text(self, plaintext: bytes):
         """Class to generate digests for plaintext files using specified hash algorithms."""
-        plaintext = []
-        with open(plaintext_file_path, "rb") as plaintext_file:
-            plaintext = plaintext_file.read()
-
         func = hashes.Hash(self._hash_func[self._hash_name], backend=default_backend())
         func.update(plaintext)
         digest = func.finalize()
-        with open(os.path.join(output_directory, "plain_text_digest.bin"), "wb") as file:
-            file.write(digest)
-        with open(os.path.join(output_directory, "plain_text_size.txt"), "w") as file:
-            file.write(str(len(plaintext)))
+        return digest, len(plaintext)
 
 
-class Encryptor:
+class Encryptor(SuitEncryptorBase):
     """Class to handle encryption operations using specified key wrap algorithms."""
 
     kms = None
 
-    def __init__(self, kw_alg: SuitKWAlgorithms):
-        """Initialize the Encryptor with a specified key wrap algorithm."""
-        if kw_alg == SuitKWAlgorithms.A256KW:
-            self.cose_kw_alg = SuitAlgorithms.COSE_ALG_A256KW.value
-        else:
-            self.cose_kw_alg = SuitAlgorithms.COSE_ALG_DIRECT.value
-        pass
-
     def init_kms_backend(self, kms_script, context):
         """Initialize the KMS from the provided script backend based on the passed context."""
         module_name = "SuitKMS_module"
         kms_module = _import_module_from_path(module_name, kms_script)
+        if not hasattr(kms_module, "suit_kms_factory"):
+            raise ValueError(f"Python script {kms_script} does not contain the required suit_kms_factory function")
         self.kms = kms_module.suit_kms_factory()
         if not isinstance(self.kms, SuitKMSBase):
             raise ValueError(f"Class {type(self.kms)} does not implement the required SuitKMSBase interface")
         self.kms.init_kms(context)
 
-    def generate_kms_artifacts(self, plaintext_file_path: Path, key_name: str, context: str):
+    def generate_kms_artifacts(self, asset_plaintext: bytes, key_name: str, context: str):
         """Generate encrypted artifacts using the key management system.
 
         This method reads the plaintext file, encrypts it using the specified key wrap algorithm,
```

Comment:
This method no longer reads the plaintext file. Please align the docs.

Reply:
Fixed